File: app/src/main/java/com/remotearthsolutions/expensetracker/utils/AlertDialogUtils.java
package com.remotearthsolutions.expensetracker.utils;

import android.content.Context;
import androidx.appcompat.app.AlertDialog;
import com.remotearthsolutions.expensetracker.contracts.BaseView;

public class AlertDialogUtils {

    public static void show(Context context, String title, String message, String btnOk, String btnCancel, BaseView.Callback callback) {
        AlertDialog.Builder builder = new AlertDialog.Builder(context);
        builder.setTitle(title);
        builder.setCancelable(false);
        builder.setMessage(message);
        builder.setPositiveButton(btnOk, (dialog, which) -> {
            if (callback != null) {
                callback.onOkBtnPressed();
            }
        });
        if (btnCancel != null) {
            builder.setNegativeButton(btnCancel, (dialog, which) -> {
                if (callback != null) {
                    callback.onCancelBtnPressed();
                }
            });
        }
        builder.create();
        builder.show();


    }
}

File: app/src/main/java/com/remotearthsolutions/expensetracker/utils/AlertDialogUtils.kt
package com.remotearthsolutions.expensetracker.utils

import android.content.Context
import android.graphics.Typeface
import com.rahman.dialog.Utilities.SmartDialogBuilder
import com.remotearthsolutions.expensetracker.R
import com.remotearthsolutions.expensetracker.contracts.BaseView

object AlertDialogUtils {
    fun show(
        context: Context?,
        title: String?,
        message: String?,
        btnOk: String?,
        btnCancel: String?,
        callback: BaseView.Callback?,
        cancellable: Boolean = false
    ) {

        SmartDialogBuilder(context!!).apply {

            setTitle(title)
            setSubTitle(message)
            setCancalable(cancellable)
            setPositiveButton(btnOk) {
                it.dismiss()
                callback?.onOkBtnPressed()

            }
            if (btnCancel != null) {
                setNegativeButtonHide(false)
                setNegativeButton(btnCancel) {
                    it.dismiss()
                    callback?.onCancelBtnPressed()
                }
            } else {
                setNegativeButtonHide(true)
            }
            build().show()
        }
    }
}